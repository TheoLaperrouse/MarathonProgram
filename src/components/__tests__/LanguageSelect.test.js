import LanguageSelector from '@/components/LanguageSelect.vue';
import { mount } from '@vue/test-utils';
import { useLocalStorage } from '@vueuse/core';
import { ref } from 'vue';
import { useI18n } from 'vue-i18n';

vi.mock('vue-i18n');
vi.mock('@vueuse/core');

describe('LanguageSelect', () => {
    let localeMock;
    let localStorageMock;

    beforeEach(() => {
        vi.resetAllMocks();

        localeMock = ref({ value: 'en' });
        useI18n.mockReturnValue({ locale: localeMock });

        localStorageMock = ref({ value: 'en' });
        useLocalStorage.mockReturnValue(localStorageMock);
    });

    it('should match snapshot', () => {
        const wrapper = mount(LanguageSelector);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('changes language and updates locale and localStorage', async () => {
        const wrapper = mount(LanguageSelector);

        const select = wrapper.find('select');
        await select.setValue('fr');

        expect(localeMock.value).toBe('fr');
        expect(localStorageMock.value).toBe('fr');
    });

    it('load the locale with the localStorage value', () => {
        localStorageMock.value = 'fr';
        localeMock.value = 'en';

        mount(LanguageSelector);

        expect(localeMock.value).toBe('fr');
    });
});
