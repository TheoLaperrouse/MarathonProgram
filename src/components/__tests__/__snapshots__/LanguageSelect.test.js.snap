// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LanguageSelect > should match snapshot 1`] = `
"<div data-v-2a5103fe="" class="relative w-40"><select data-v-2a5103fe="" class="w-full px-4 py-2">
    <option data-v-2a5103fe="" value="en">English</option>
    <option data-v-2a5103fe="" value="fr">Fran√ßais</option>
  </select>
  <div data-v-2a5103fe="" class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2"><svg data-v-2a5103fe="" class="h-5 w-5 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
      <path data-v-2a5103fe="" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M19 9l-7 7-7-7"></path>
    </svg></div>
</div>"
`;
